plugins/modules/files/archive.py

```diff
@@ -23,6 +23,7 @@
     description:
       - Remote absolute path, glob, or list of paths or globs for the file or files to compress or archive.
     type: list
+    elements: str
```

Comment:
```suggestion
    elements: path
```
Same for the other (and the `elements` in `path`).

Setting both to `path` also allows to remove the `os.path.expanduser` and `os.path.expandvars` calls, and that allows to remove the `use-argspec-type-path` ignore.

Reply:
Thanks for the help! Fixed in https://github.com/ansible-collections/community.general/pull/1039/commits/8a31be83468801b7ae8bd59eabdbc98c458e1f3f.